boa_engine/src/builtins/array/mod.rs

```diff
@@ -1519,38 +1526,20 @@ impl Array {
         // 2. Let len be ? LengthOfArrayLike(O).
         let len = o.length_of_array_like(context)?;
 
-        // 3. If IsCallable(predicate) is false, throw a TypeError exception.
         let predicate = args.get_or_undefined(0).as_callable().ok_or_else(|| {
```

Comment:
This check is now obsolete since `find_via_predicate` checks if the `predicate.is_callable()`, but kept it for the completer error message in all variants. Let me know if this should be removed anyways.

Reply:
I wouldn't do this extra check, since the spec doesn't have it.

Reply:
I agree that the check is not needed, but it would be nice to have better error messages (especially since we don't have line numbers, yet).

Maybe we could have a parameter `caller_name: &str` on find_via_predicate that pass the caller `"Array.prototype.find"` and and construct the error message. (`format!("{caller_name}": predicate not a function)`).

Reply:
Seems like a good solution thanks, implemented in fd1af53